### ISSUE TITLE
Only version info from version catalog can be used in buildscript block

### ISSUE THREAD

**Issue by strindberg** [gradle/gradle#18105]
### Expected Behavior

Libraries and bundles defined in a version catalog should be usable within the buildscript block of a gradle build file.

### Current Behavior

Libraries and bundles defined in a version catalog (in toml format) can not be found within the buildscript block of a gradle build file, only versions defined in this way can be used.

### Context

We are trying to consolidate our dependencies in the way outlined in the version catalog documentation. We have dependencies that are used both within and outside the buildscript block, and want to make sure that they are using the same version.

### Steps to Reproduce 

Assume the following `libs.versions.toml`:

```toml
[versions]
jooq = "3.15.1"

[libraries]
jooq-core = { module = "org.jooq:jooq", version.ref = "jooq" }
jooq-codegen = { module = "org.jooq:jooq-codegen", version.ref = "jooq" }
```

We want to use the library definition in our `build.gradle.kts` like so:

```kotlin
buildscript {
    configurations.classpath {
        resolutionStrategy {
            setForcedModules(libs.jooq.codegen)
        }
    }
}
```

But this results in an error.

This works:

```kotlin
buildscript {
    configurations.classpath {
        resolutionStrategy {
            setForcedModules("org.jooq:jooq-codegen:${libs.versions.jooq.get()}")
        }
    }
}
```
In other words: the versions defined in the toml file can be used within the buildscript block, but not the libraries.

The defined libraries can be used without problems in the dependeny section of the build file:

```kotlin
dependencies {
    ...
    implementation(libs.jooq.core)
    ....
}
```

### Your Environment

Gradle version 7.2, run on Mac.

Build scan URL:

**Comment by mlopatkin** [comment]
Sorry that you're having trouble with Gradle!

Your issue lacks information about how to reproduce the problem you're having. A reproducer project can really help us track down and fix your problem quicker.  We may also be able to suggest workarounds or ways to avoid the problem if we can reproduce it. It would be great to see the error message that you're seeing.

I suspect that the problem you're facing is a duplicate of the #17874. Version catalog accessors are of special type and not all places that accept dependency notation accept this type yet.

This issue will be closed after 7 days unless you can provide more information.

**Comment by strindberg** [comment]
That's right, it does indeed look like a duplicate of #17874. I'll close this one.
